code/go/0chain.net/smartcontract/dbs/event/blobber_aggregate.go

```diff
@@ -0,0 +1,201 @@
+package event
+
+import (
+	"math"
+
+	"0chain.net/smartcontract/common"
+	"github.com/0chain/common/core/currency"
+	"github.com/0chain/common/core/logging"
+	"go.uber.org/zap"
+	"gorm.io/gorm"
+	"gorm.io/gorm/clause"
+)
+
+type BlobberAggregate struct {
+	gorm.Model
+
+	BlobberID string `json:"blobber_id" gorm:"index:idx_blobber_aggregate,unique"`
+	Round     int64  `json:"round" gorm:"index:idx_blobber_aggregate,unique"`
+
+	WritePrice   currency.Coin `json:"write_price"`
+	Capacity     int64         `json:"capacity"`  // total blobber capacity
+	Allocated    int64         `json:"allocated"` // allocated capacity
+	SavedData    int64         `json:"saved_data"`
+	ReadData     int64         `json:"read_data"`
+	OffersTotal  currency.Coin `json:"offers_total"`
+	UnstakeTotal currency.Coin `json:"unstake_total"`
+	TotalStake   currency.Coin `json:"total_stake"`
+
+	TotalServiceCharge  currency.Coin `json:"total_service_charge"`
+	ChallengesPassed    uint64        `json:"challenges_passed"`
+	ChallengesCompleted uint64        `json:"challenges_completed"`
+	OpenChallenges      uint64        `json:"open_challenges"`
+	InactiveRounds      int64         `json:"InactiveRounds"`
+	RankMetric          float64       `json:"rank_metric" gorm:"index:idx_ba_rankmetric"`
+}
+
+func (edb *EventDb) ReplicateBlobberAggregate(p common.Pagination) ([]BlobberAggregate, error) {
+	var snapshots []BlobberAggregate
+
+	queryBuilder := edb.Store.Get().
+		Model(&BlobberAggregate{}).Offset(p.Offset).Limit(p.Limit)
+
+	queryBuilder.Order(clause.OrderByColumn{
+		Column: clause.Column{Name: "id"},
+		Desc:   false,
+	})
+
+	result := queryBuilder.Scan(&snapshots)
+	if result.Error != nil {
+		return nil, result.Error
+	}
+
+	return snapshots, nil
+}
+func (edb *EventDb) updateBlobberAggregate(round, pageAmount int64, gs *globalSnapshot) {
+	count, err := edb.GetBlobberCount()
+	if err != nil {
+		logging.Logger.Error("update_blobber_aggregates", zap.Error(err))
+		return
+	}
+	size, currentPageNumber, subpageCount := paginate(round, pageAmount, count)
+
+	exec := edb.Store.Get().Exec("CREATE TEMP TABLE IF NOT EXISTS temp_ids "+
+		"ON COMMIT DROP AS SELECT blobber_id as id FROM blobbers ORDER BY (blobber_id, creation_round) LIMIT ? OFFSET ?",
+		size, size*currentPageNumber)
+	if exec.Error != nil {
+		logging.Logger.Error("error creating temp table", zap.Error(exec.Error))
+		return
+	}
+
+	for i := 0; i < subpageCount; i++ {
+		edb.calculateBlobberAggregate(gs, round, pageLimit, int64(i)*pageLimit)
+	}
+
+}
+
+func paginate(round int64, pageAmount int64, count int64) (int64, int64, int) {
+	size := int64(math.Ceil(float64(count) / float64(pageAmount)))
+	currentPageNumber := round % pageAmount
+
+	currentPageSize := size
+	if pageAmount-currentPageNumber == 1 { //last page
+		currentPageSize = count - size*currentPageNumber
```

Comment:
yeah, good catch! we don't need this adjust for the last page anymore, since with 
```go
	size := int64(math.Ceil(float64(count) / float64(pageAmount)))
```
we guaranteed to have enough space to paginate count to exactly `pageAmount` pages